raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -306,10 +314,22 @@ private VoteResponseData buildVoteResponse(Errors error, boolean voteGranted) {
                 .setVoteGranted(voteGranted);
     }
 
-    private VoteResponseData handleVoteRequest(VoteRequestData request) throws IOException {
-        Optional<Exception> errorOpt = validateLeaderOnlyRequest(request.candidateId(), request.candidateEpoch());
+    /**
+     * Handle a Vote request. This API may return the following errors:
+     *
+     * - {@link Errors#BROKER_NOT_AVAILABLE} if this node is currently shutting down
+     * - {@link Errors#FENCED_LEADER_EPOCH} if the epoch is smaller than this node's epoch
+     * - {@link Errors#INCONSISTENT_VOTER_SET} if the request suggests inconsistent voter membership (e.g.
+     *      if this node or the sender is not one of the current known voters)
+     * - {@link Errors#INVALID_REQUEST} if the last epoch or offset are invalid
```

Comment:
My reasoning is that this error is reserved for cases that are not expected to happen with a proper functioning application. So whenever we see `INVALID_REQUEST`, it indicates a bug. The case of `INCONSISTENT_VOTER_SET` is a bit different because we could definitely see this in the middle of a reassignment (once we've added supported).